reactor-support/src/main/java/tech/picnic/reactor/ReactorSpring4Util.java

```diff
@@ -0,0 +1,194 @@
+package tech.picnic.reactor;
+
+import com.google.common.annotations.VisibleForTesting;
+import java.io.IOException;
+import java.io.UncheckedIOException;
+import java.time.Duration;
+import java.util.List;
+import java.util.function.Function;
+import java.util.function.LongFunction;
+import javax.annotation.Nullable;
+import org.reactivestreams.Publisher;
+import org.springframework.http.MediaType;
+import org.springframework.web.context.request.async.DeferredResult;
+import org.springframework.web.servlet.mvc.method.annotation.SseEmitter;
+import reactor.core.publisher.Flux;
+import reactor.core.scheduler.Scheduler;
+import reactor.core.scheduler.Schedulers;
+
+/**
+ * Utility class providing operations to convert Reactor reactive types into asynchronous Spring 4.x
+ * web types.
+ */
+@SuppressWarnings("NoFunctionalReturnType")
```

Comment:
This is needed here as the `Function`s returned here are the native `java.util.function.Function`s which are functional interfaces as opposed to the `io.reactivex.functions.Function`.